Automattic/wp-calypso
# Move WebPreview to calypso-ui package

#### Changes proposed in this Pull Request

* Moves the `WebPreview` component and necessary dependencies to the `calypso-ui` package.
* Refactors the `WebPreview` component and separates device logic.
* Moves the `AsyncLoad` component to `calypso-ui`.
* Moves the `SpinnerLine` component to `calypso-ui`.

#### Thoughts
* There's one test still failing for translation.  Not sure if this has something to do with the default `translate` prop added here or another error.  Any feedback here appreciated!
* In order to keep the scope down on this PR, I chose to separate out the toolbar from the `WebPreview` component.  While some of this logic could benefit from being shared across projects, there are many dependencies tied up in Calypso.
* Similarly, the touch-detect `hasTouch()` method was manually copied over to avoid creating a new `calypso-lib` package.
* Devices and logic around the SEO panel were separated from the main component since these will probably vary widely from project to project.

#### Testing instructions

* Visit all the areas around the site that use the `WebPreview` component and make sure they continue to work as expected with correct buttons/content appearing based on props.
* Make sure that toggling the device in the toolbar continues to work as expected (especially when toggling between custom preview content types like `seo`).
* Make sure that the `SpinnerLine` continues to work across the site.
* Make sure the inline help popover does not lose focus if open before loading a new URL in the iframe.
* Optionally: `npm link` the calypso-ui package, bump the version, and test that you can import the `WebPreview` component in another project.

Fixes #34295
